src/services/activity.service.ts

```diff
@@ -278,17 +300,20 @@ export class ActivityService {
       }
     }
    */
-   mergeReferenceToSequence(activity) {
-    let refs = this.rebuildReferences(activity.References);
-
+   mergeReferenceToSequence(activity): Object {
     // @NOTE: first "[0]" sequence is the assessment of an activity
     let sequence = (activity.ActivitySequence) ? activity.ActivitySequence[0] : {};
 
-    if (!_.isEmpty(sequence)) {
-      // activity.ActivitySequence.forEach(seq => {
-        let modelId = sequence.model_id;
-        sequence.context_id = refs[modelId];
-      // });
+    // `References` object is optional (modified on 6 October 2017)
+    if (activity.References) {
+      let refs = this.rebuildReferences(activity.References);
+      if (!_.isEmpty(sequence)) {
+        // @NOTE: API only support first ActivitySequence atm
+        // activity.ActivitySequence.forEach(seq => {
```

Comment:
Hi @trtshen, do we need to remove this commented code?

Reply:
@dameng03 @DamengRandom that is retained as a reference to comment at line 307.
Let's keep this as a reference for future Practera API changes.

Reply:
Yup sure @trtshen